test/inductor/test_fused_attention.py

```diff
@@ -165,6 +165,42 @@ def sfdp_pattern_6(query, key, value):
 
         self._check_common(sfdp_pattern_6, contains=False)
 
+    def test_pattern_fails_with_tensor_factor(self):
+        # https://github.com/pytorch/pytorch/issues/99124
+        class Model(torch.nn.Module):
+            def __init__(self, is_inv_factor):
+                super(Model, self).__init__()
+                self.scale_factor = torch.nn.Parameter(
+                    torch.ones(4, 1, 1, device="cuda")
+                )
+                self.is_inv_factor = is_inv_factor
+
+            def forward(self, query, key, value) -> torch.Tensor:
+                y = torch.matmul(query, key.transpose(-2, -1))
+                if self.is_inv_factor:
+                    y = y.div(self.scale_factor)
+                else:
+                    y = y.mul(self.scale_factor)
+                return y.softmax(dim=-1).matmul(value)
+
+        tensor_shape = (2, 4, 4, 4)
+        args = [
+            torch.randn(tensor_shape, device="cuda"),
+            torch.randn(tensor_shape, device="cuda"),
+            torch.randn(tensor_shape, device="cuda"),
+        ]
+        with torch.no_grad():
```

Comment:
Can you test training as well?

Reply:
training test is added, please note that even if not have this PR, the training path can work well(the pattern doesn't match), and has an accuracy gap compared with eager mode.